components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +74,11 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public void resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        resetRoute.reset();
```

Comment:
Hi Claus, 
I added a null check as below, 

if (resetRoute != null) {
            resetRoute.reset(true);
        } 

However, if the user uses some incorrect id of the route which does not match as expected, for example, like below,

curl -XPOST -s http://localhost:8080/camel/routes/info/reset

Do we throw any exception ? Like the code below, 

if (resetRoute != null) {
            resetRoute.reset(true);
        } else {
            throw new GenericBeansException("The route id does not match : " + resetRoute.getCamelId());
        }

And then, i get

[vgohel@dhcp193-148 camel-example-spring-boot-routecontroller]$ curl -XPOST -s http://localhost:8080/camel/routes/info/reset

{"timestamp":1501134317055,"status":500,"error":"Internal Server Error","exception":"org.apache.camel.spring.boot.actuate.endpoint.CamelRoutesMvcEndpoint$GenericException","message":"Internal Server Error","path":"/camel/routes/info/reset"}

Let me know about this. I feel its not right to throw an exception. But if the 'id' is incorrect, the command then does not do anything.